### PR TITLE
Isaac solution

### PR DESCRIPTION
**Pull request by Crunchwrap78**
comfort level 3
completeness 5

Making the app is one thing, but debugging is a nightmare.....
errors... errors every...

### CODE REVIEW

**Comment by Crunchwrap78** [comment]
on the positive side I got all bonuses

File: index.html [hunk 1]
```diff
@@ -1,15 +1,31 @@
...
+    <script src="js/songs/song.factory.js"></script>
+    <script src="js/songs/player.directive.js"></script>
+    <script src="js/songs/form.directive.js"></script>
+    <script src="js/songs/songsapi.js"></script>
   </head>
   <body>
     <header>WDI Radio</header>
-
+    <form id="search">
+      <input type="search" id="search-keyword">
```
Comment (jshawl) [inline index.html]: should have ng-model, no?

File: js/songs/edit.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<h2>Edit Song</h2>
+
+<div>
+  <song-form data-song='SongEditViewModel.song'></song-form>
```
Comment (jshawl) [inline js/songs/edit.html]: excellent use of directive!!!

File: js/songs/form.directive.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+"use strict";
+(function(){
+  angular
+  .module("songs")
+  .directive('songForm', [
+    "SongFactory",
+    "$state",
+    GrumbleFormDirectiveFunction
+  ]);
+  function GrumbleFormDirectiveFunction(SongFactory, $state){
```
Comment (jshawl) [inline js/songs/form.directive.js]: grumble?!

File: js/songs/form.directive.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    return {
+      templateUrl: 'js/songs/_song_form.html',
+      replace: true,
+      scope: {
+        song: "=",
+        formType: '@'
```
Comment (jshawl) [inline js/songs/form.directive.js]: :+1:

File: js/songs/songsapi.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+// $(document).ready(function(){
+//   $(".search").on("click", apiCall)
```
Comment (jshawl) [inline js/songs/songsapi.js]: let me know if you'd like to work through this bit together

**Comment by jshawl** [comment]
:+1: Angular errors are difficult! nice job on this assignment

**Pull request closed without merging**